rfcs/999-off-the-shelf-charts.adoc

```diff
@@ -0,0 +1,42 @@
+
+== Abstract
+
+In the integration design of Helm packages in Shipper, we've opted to have a tight contract for production, load balance Kubernetes Service manifests. This designed proved to be sub-optimal; it requires that special Helm packages *must* be created in order to be compatible with Shipper.
+
+This RFC proposes formally specifying a compatibility layer in Shipper to make it possible to use publicly available charts with it.
+
+== Rationale
+
+Shipper's contract with Helm charts specify that a chart needs to contain *one and only one* deployment manifest and *one and only one* load balancing service manifest. Having multiple Kubernetes service is useful for, for example, enable access to pods belonging only to the new version of an image *before* starting a release process.
+
+To identify this load balancing service manifest, once it is installed on a Kubernetes cluster, a label on the specific load balancing service is used (`shipper-lb: production` more specifically). This choice made any existing chart on the wild incompatible with Shipper.
+
+== Specification
+
+To overcome that limitation, we want to introduce the field `.spec.template.loadBalancer` field in the Application CRD; this field will be used to propagate relevant load balancing configuration to Shipper.
+
+The following listing shows an Application manifest with the `.spec.template.loadBalancer`:
+
+..spec.template.loadBalancer example
+[source,yaml]
+----
+spec:
+  loadBalancer:     # <1>
+    kubernetes:     # <2>
```

Comment:
Something I was pondering: we should make this key as specific as we can in order to have some forward compatibility. For example, k8s might introduce some awesome new load balancing thing (IngressNG, for example), and if we name this key “kubernetes”, we won’t be able to add an implementation for that without inventing another vanilla Kubernetes key.

 So a really rough sketch might be:
```
loadBalancer:
  kubernetes:
    core/v1/service:
      name: foobar
      selector: {}
```

Looking at it, you might not need ‘kubernetes’ as a level anymore, since GVK is unambiguous. 

Same principle applies to Istio if they change their APIs around. However, it does make our syntax more verbose, which isn’t super exciting.